quixstreams/sources/community/gcp_pubsub/gcp_pubsub.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+from proto.datetime_helpers import DatetimeWithNanoseconds
+
+from quixstreams.models import Topic
+from quixstreams.sources import Source
+
+from .components import GCPPubSubConfig
+from .components.gcp_consumer import GCPPubSubConsumer
+
+__all__ = ("GCPPubSubSource",)
+
+
+logger = logging.getLogger(__name__)
+
+
+class GCPPubSubSource(Source):
+    """
+    This source enables reading from a Google Cloud Platform (GCP) PubSub topic,
+    dumping it to a kafka topic using desired SDF-based transformations.
+
+    Provides "at-least-once" guarantees.
+
+    Currently, forwarding message keys ("ordered messages" in GCP) is unsupported.
```

Comment:
Not sure if this is true. Or at least slightly inconsistent.

* Line 104 has `key=message.ordering_key or None,` so we suggest that we do forward ordering_key.
* Of course only if subscription was not created by this connector in its current form
  * because we don't set `enable_message_ordering`, so `ordering_key` will always be.. err empty string?

Reply:
Yeah you're right, technically it could support it if the subscription was not created by this application. 

I'll document this appropriately :joy: